power10dan/tldrResearchPaper
# Redux/React front end Upload File

# What does this PR do
This PR adds working upload file calls to the backend server via react and redux. It also implements a lot of quality of life changes: non-logged in users are unable to access the uploadFile and about buttons, the app will successfully display error messages received from the server as well.

# How to test
The only real way to test this PR is going to be to load up GROBID, load up Django, and load our front end and then do the following:
1) Create a new user, this should automatically log you in
2) Click Upload File
3) Upload a FIle and observe the output of each

Here is the output from my machine:
Django:
```
Performing system checks...

System check identified no issues (0 silenced).
November 08, 2017 - 17:12:24
Django version 1.11.5, using settings 'uploads.settings.settings_dev'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
[08/Nov/2017 17:12:42] "OPTIONS /rest-auth/registration/ HTTP/1.1" 200 0
[08/Nov/2017 17:12:42] "POST /rest-auth/registration/ HTTP/1.1" 201 50
[08/Nov/2017 17:12:42] "OPTIONS /login/ HTTP/1.1" 200 0
[08/Nov/2017 17:12:42] "POST /login/ HTTP/1.1" 200 173
[08/Nov/2017 17:12:51] "OPTIONS /api/uploadFile/Applicative.pdf HTTP/1.1" 200 0
Grobid Working
Grobid Finished
[08/Nov/2017 17:13:35] "POST /api/uploadFile/Applicative.pdf HTTP/1.1" 204 0
```

GROBID:
```
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/AppBackEnd/lib/grobid-core-0.4.4-SNAPSHOT.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/doyougnu/.m2/repository/org/slf4j/slf4j-log4j12/1.6.6/slf4j-log4j12-1.6.6.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/fulltext/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/fulltext/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/segmentation/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/segmentation/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/figure/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/figure/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/header/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/header/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/name/header/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/name/header/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/name/citation/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/name/citation/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/affiliation-address/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/affiliation-address/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/citation/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/citation/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/reference-segmenter/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/reference-segmenter/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/date/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/date/model.wapiti
[Wapiti] Loading model: "/home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/table/model.wapiti"
Model path: /home/doyougnu/School_Files/Hoops_17-18/Fall/SE/tldrResearchPaper/grobid-grobid-parent-0.4.4/grobid-home/models/table/model.wapiti
```
